src/taker/api.rs

```diff
@@ -286,6 +286,31 @@ impl Taker {
         })
     }
 
+    fn sweep_incoming_swapcoins(&mut self, feerate: f64) -> Result<(), TakerError> {
+        // Convert each IncomingSwapCoin to the required tuple.
+        let coins_to_spend = self
+            .ongoing_swap_state
+            .incoming_swapcoins
+            .iter()
+            .map(|swapcoin| swapcoin.to_utxo_tuple())
+            .collect::<Vec<_>>();
```

Comment:
You need be very careful here. Only select those incoming swap coins for which other's privkey is known, ie. swap has completed.